plugins/huaweicloud/src/main/java/org/eclipse/xpanse/plugins/huaweicloud/manage/HuaweiCloudVmStateManager.java

```diff
@@ -0,0 +1,112 @@
+        this.huaweiCloudManagerClient = huaweiCloudManagerClient;
+        this.huaweiCloudServerManageRequestConverter = huaweiCloudServerManageRequestConverter;
+    }
+
+    /**
+     * Start the Huawei Cloud Ecs server.
+     */
+    public boolean startService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchStartServersRequest request =
+                huaweiCloudServerManageRequestConverter.buildBatchStartServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchStartServersResponse response = ecsClient.batchStartServers(request);
+        return checkEcsExecResultByJobId(ecsClient, response.getJobId());
+    }
+
+    /**
+     * Stop the Huawei Cloud Ecs server.
+     */
+    public boolean stopService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchStopServersRequest batchStopServersRequest =
+                huaweiCloudServerManageRequestConverter.buildBatchStopServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchStopServersResponse response = ecsClient.batchStopServers(batchStopServersRequest);
+        return checkEcsExecResultByJobId(ecsClient, response.getJobId());
+    }
+
+    /**
+     * Restart the Huawei Cloud Ecs server.
+     */
+    public boolean restartService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchRebootServersRequest request =
+                huaweiCloudServerManageRequestConverter.buildBatchRebootServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchRebootServersResponse response = ecsClient.batchRebootServers(request);
+        return checkEcsExecResultByJobId(ecsClient, response.getJobId());
+    }
+
+    private EcsClient getEcsClient(ServiceManagerRequest serviceManagerRequest) {
+        AbstractCredentialInfo credential =
+                credentialCenter.getCredential(Csp.HUAWEI, CredentialType.VARIABLES,
+                        serviceManagerRequest.getUserId());
+        ICredential icredential = huaweiCloudManagerClient.getCredential(credential);
+        return huaweiCloudManagerClient.getEcsClient(icredential,
+                serviceManagerRequest.getRegionName());
+    }
+
+    private boolean checkEcsExecResultByJobId(EcsClient ecsClient, String jobId) {
+        ShowJobResponse response = ecsClient.showJobInvoker(new ShowJobRequest().withJobId(jobId))
+                .retryTimes(RETRY_TIMES)
+                .retryCondition((resp, ex) -> Objects.nonNull(resp) && !resp.getStatus()
+                        .equals(ShowJobResponse.StatusEnum.SUCCESS))
+                .backoffStrategy(new SdkBackoffStrategy(BASE_DELAY, MAX_BACK_OFF_IN_MILLISECONDS))
+                .invoke();
```

Comment:
here can you log the response ShowJobResponse.failReason and ShowJobResponse.message whenver the status is not SUCCESS.
we must log with log level error.